ITGlueAPI/Internal/APIKey.ps1

```diff
@@ -1,35 +1,227 @@
+
+#>
+
+    [cmdletbinding()]
+    Param ()
+
+    if ($ITGlue_API_Key) {
+        $ITGlue_API_Key
+    }
+    else{
+        Write-Warning "No API key exists. Please run Add-ITGlueAPIKey to add one."
     }
+
 }
 
+
+
 function Remove-ITGlueAPIKey {
-    Remove-Variable -Name "ITGlue_API_Key" -Scope global -Force
+<#
+    .SYNOPSIS
+        Removes the ITGlue API key
+
+    .DESCRIPTION
+        The Remove-ITGlueAPIKey cmdlet removes the ITGlue API key from
+        global variable.
+
+    .EXAMPLE
+        Remove-ITGlueAPIKey
+
+        Removes the ITGlue API key global variable.
+
+    .NOTES
+        N\A
+
+    .LINK
+        https://github.com/itglue/powershellwrapper
+
+#>
+
+    [cmdletbinding(SupportsShouldProcess)]
+    Param ()
+
+    if ($ITGlue_API_Key) {
+        Remove-Variable -Name "ITGlue_API_Key" -Scope global -Force -WhatIf:$WhatIfPreference
+    }
+    else{
+        Write-Warning 'The ITGlue API key variable is not set. Nothing to remove'
+    }
 }
 
-function Get-ITGlueAPIKey {
-    if($null -eq $ITGlue_API_Key) {
-        Write-Error "No API key exists. Please run Add-ITGlueAPIKey to add one."
+
+
+function Test-ITGlueAPIKey {
+<#
+    .SYNOPSIS
+        Test the ITGlue API key
+
+    .DESCRIPTION
+        The Test-ITGlueAPIKey cmdlet tests the base URI & API key that are defined
+        in the Add-ITGlueBaseURI & Add-ITGlueAPIKey cmdlets.
+
+        Helpful when needing to validate general functionality or when using
+        RMM deployment tools
+
+        The ITGlue Regions endpoint is called in this test
+
+    .PARAMETER base_uri
+        Define the base URI for the ITGlue API connection
+        using ITGlue's URI or a custom URI.
+
+        By default the value used is the one defined by Add-ITGlueBaseURI function
+            'https://api.itglue.com'
+
+    .EXAMPLE
+        Test-ITGlueApiKey
+
+        Tests the base URI & API key that are defined in the
+        Add-ITGlueBaseURI & Add-ITGlueAPIKey cmdlets.
+
+    .EXAMPLE
+        Test-ITGlueApiKey -base_uri http://myapi.gateway.example.com
+
+        Tests the defined base URI & API key that was defined in
+        the Add-ITGlueAPIKey cmdlet.
+
+        The full base uri test path in this example is:
+            http://myapi.gateway.example.com/regions
+
+    .NOTES
+        N\A
+
+    .LINK
+        https://github.com/itglue/powershellwrapper
+
+#>
+
+    [cmdletbinding()]
+    Param (
+        [parameter(Mandatory = $false)]
+        [ValidateNotNullOrEmpty()]
+        [string]$base_uri = $ITGlue_Base_URI
+    )
+
+    $resource_uri = "/regions"
+
+    Write-Verbose "Testing API key against [ $($base_uri + $resource_uri) ]"
+
+    try {
+
+        $ITGlue_Headers.Add('x-api-key', (New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList 'N/A', $ITGlue_API_Key).GetNetworkCredential().Password)
+
+        $rest_output = Invoke-WebRequest -Method Get -Uri ($base_uri + $resource_uri) -Headers $ITGlue_Headers -ErrorAction Stop
```

Comment:
That is a good question, the reason I used `Invoke-WebRequest` instead of `Invoke-RestMethod` here was because `Invoke-WebRequest` returns header information. I'll do some testing this weekend to see what I can do. 

As a thought what if the `Invoke-ITGlueRequest` is used to test first and if it fails then the `Invoke-WebRequest` method is called to get the failure details?

I also wouldn't be offended if we didn't include this, it is just a nice to have IMO.

Reply:
It's fine to keep it like this, since there's a good reason for doing so. Thanks for pointing that out!

Reply:
Sounds good to me, is there anything else I should clean up with this pull request? I believe the remaining conflicts are just line breaks.

Reply:
I can't think of anything, other than fixing the remaining conflicts. @adrianwells, could you do a "squash and merge" when you merge this PR? There's 11 commits now and it's a lot of clutter that probably shouldn't be in the mainline Git history.

Reply:
@Celerium Thank you. It appears more than line breaks creating a conflict in OrganizationTypes.ps1 (the other 2 files, ITGlueAPI.psm and PasswordCategories.ps1 seem to only line break issues). You able to review and update? @davidhaymond I think is close to ready.

Reply:
@adrianwells, I could be wrong but the PasswordCategories.ps1 conflict appears to be the help section.